.github/workflows/deploy.devnet.yml

```diff
@@ -0,0 +1,164 @@
+---
+name: DevNet Workflow
+
+on:  # yamllint disable-line rule:truthy
+  workflow_dispatch:
+  push:
+    branches:
+      - develop
+
+permissions:
+  id-token: write
+  contents: read
+  security-events: write
+
+jobs:
+  gosec:
+    name: Gosec and Publish
+    uses: 0xpolygon/polygon-edge/.github/workflows/security.yml@develop
+
+  build:
+    name: Build
+    uses: 0xpolygon/polygon-edge/.github/workflows/build.yml@develop
+
+  test:
+    name: Test
+    uses: 0xpolygon/polygon-edge/.github/workflows/test.yml@develop
+    needs: build
+
+  deploy_devnet:
+    name: Deploy DevNet
+    needs: build
+    concurrency: deploy_devnet
+    environment:
+      name: devnet
+      url: https://rpc.us-east-1.testing.psdk.io/
+    runs-on: ubuntu-latest
+    steps:
+      - name: Download Polygon Edge Artifact
+        uses: actions/download-artifact@v3
+        with:
+          name: polygon-edge
+
+      - name: Configure AWS Credentials
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ secrets.AWS_ROLE_ARN }}
+          aws-region: ${{ secrets.AWS_REGION }}
+
+      - name: Notify Slack
+        uses: slackapi/slack-github-action@v1.22.0
+        if: false
+        continue-on-error: true
+        env:
+          SLACK_WEBHOOK_URL: ${{ secrets.SLACK_DEPLOYMENT_WEBHOOK_URL }}
+          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+        with:
+          payload: |
+            {
+              "blocks": [
+                {
+                  "type": "header",
+                  "text": {
+                    "type": "plain_text",
+                    "text": "Devnet Deployment - Started"
+                  }
+                },
+                {
+                  "type": "section",
+                  "text": {
+                    "type": "mrkdwn",
+                    "text": "Devnet Deployment Status: ${{ job.status }}\n<${{ github.server_url }}/${{ github.repository }}/actions/runs/${{ github.run_id }}|Job Status>"
+                  }
+                }
+              ]
+            }
+
+      - name: Publish Binary package to S3
+        run: |
+          aws s3 cp ./polygon-edge.tar.gz s3://${{ secrets.POLYGON_EDGE_ARTIFACT_BUCKET }} --metadata "{\"commit_sha\":\"${GITHUB_SHA}\"}"
```

Comment:
Are we going to maintain historical versions of the package?

Reply:
[This line ](https://github.com/0xPolygon/polygon-edge/pull/765/files/30de509c23441317ff92cd7c7db9f290ee8949ea#diff-5b99851af7508b7aa954e62ebb2d8ae12f67239464e467fff03fe1a7994056eaR91-R92) sends an SSM command  that stops the polygon-edge service and removes the node's chain data.

 TestNet deployment does not run the cleanup command.

Reply:
For `devnet` and `testnet`, we can use S3 versions with the `commit_sha` metadata. I am open to discuss on a more formalized approach for artifacting here.

Reply:
S3 versioning should be 👌  at this point.